### ISSUE TITLE
NullPointerException in AtmospherePushConnection.sendMessage()

### ISSUE THREAD

**Issue by vaadin-bot** [vaadin/framework#5776]
**Originally by _archie172_**

___
Vaadin 7.2.7.

Got this NPE:
```
java.lang.RuntimeException: Push failed
        at com.vaadin.server.communication.AtmospherePushConnection.push(AtmospherePushConnection.java:158)
        at com.vaadin.server.communication.AtmospherePushConnection.push(AtmospherePushConnection.java:133)
        at com.vaadin.ui.UI.close(UI.java:1176)
        at com.vaadin.server.VaadinSession.addUI(VaadinSession.java:1194)
        at com.vaadin.server.communication.UIInitHandler.getBrowserDetailsUI(UIInitHandler.java:224)
        at com.vaadin.server.communication.UIInitHandler.synchronizedHandleRequest(UIInitHandler.java:74)
        at com.vaadin.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:41)
        at com.vaadin.server.VaadinService.handleRequest(VaadinService.java:1401)
        at com.vaadin.server.VaadinServlet.service(VaadinServlet.java:237)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:728)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:305)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
        at org.dellroad.stuff.servlet.ExceptionLoggingFilter.doFilterInternal(ExceptionLoggingFilter.java:38)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:243)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:222)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:123)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:502)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:171)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:99)
        at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:953)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:118)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:408)
        at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1023)
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:589)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1686)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:744)
Caused by: java.lang.NullPointerException
        at com.vaadin.server.communication.AtmospherePushConnection.sendMessage(AtmospherePushConnection.java:173)
        at com.vaadin.server.communication.AtmospherePushConnection.push(AtmospherePushConnection.java:156)
        ... 29 more
```

followed by this:

```
java.lang.NullPointerException
        at com.vaadin.server.communication.AtmospherePushConnection.disconnect(AtmospherePushConnection.java:266)
        at com.vaadin.ui.UI.setPushConnection(UI.java:1519)
        at com.vaadin.ui.UI.setSession(UI.java:432)
        at com.vaadin.server.VaadinSession.removeUI(VaadinSession.java:893)
        at com.vaadin.server.VaadinService$2.run(VaadinService.java:1181)
        at com.vaadin.ui.UI.accessSynchronously(UI.java:1309)
        at com.vaadin.server.VaadinService.removeClosedUIs(VaadinService.java:1176)
        at com.vaadin.server.VaadinService.cleanupSession(VaadinService.java:1140)
        at com.vaadin.server.VaadinService.requestEnd(VaadinService.java:1343)
        at com.vaadin.server.VaadinService.handleRequest(VaadinService.java:1415)
        at com.vaadin.server.VaadinServlet.service(VaadinServlet.java:237)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:728)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:305)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
        at org.dellroad.stuff.servlet.ExceptionLoggingFilter.doFilterInternal(ExceptionLoggingFilter.java:38)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:243)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:222)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:123)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:502)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:171)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:99)
        at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:953)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:118)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:408)
        at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1023)
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:589)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1686)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:744)
```

This also resulted in my UI root's `detach()` method being invoked a more than once, which led to additional exceptions because it was not expecting that to ever occur.

___
 Imported from https://dev.vaadin.com/ issue #14859

**Comment by andreika63** [comment]
some ui element has its json representation, when value of some json field is null:
vaadin-shared-8.8.5-sources.jar!/elemental/json/impl/JsonUtil.java:206
elemental.json.impl.JsonUtil#quote
produces NPE

**Comment by TatuLund** [comment]
@andreika63 I do not see the connection between your comment and reported issue. The original issue report is also too vague, to work on.

**Comment by TatuLund** [comment]
This issue is fixed in Vaadin 7.7.38. and Vaadin 8.20.0.

**Comment by PAX523** [comment]
I can't confirm that. This is not necessarily an issue of Vaadin. It's actually possible that it's caused by another general network-related problem. But it would be good to know what.

Before you ask: my company has prime subscription, and we already were discussing it with you Tatu, with no reasonable result.
